polkadot/node/network/approval-distribution/src/lib.rs

```diff
@@ -2083,44 +2300,175 @@ pub const MAX_APPROVAL_BATCH_SIZE: usize = ensure_size_not_zero(
 	MAX_NOTIFICATION_SIZE as usize / std::mem::size_of::<IndirectSignedApprovalVote>() / 3,
 );
 
+// Low level helper for sending assignments.
+async fn send_assignments_batched_inner(
+	sender: &mut impl overseer::ApprovalDistributionSenderTrait,
+	batch: impl IntoIterator<Item = (IndirectAssignmentCertV2, CandidateBitfield)>,
+	peers: &[PeerId],
+	peer_version: ValidationVersion,
+) {
+	let peers = peers.into_iter().cloned().collect::<Vec<_>>();
```

Comment:
This seems wasteful. Why take a &[], if we actually need a Vec? I immediately found at least one call side, where taking a vec directly would have avoided that cloning.

Reply:
Will look into optimising it.

Reply:
It appears as wasteful, but it is actually, because in all places where it gets called it is in a loop so we still need a clone because `NetworkBridgeTxMessage::SendValidationMessage` requires a Vec<PeerId>, anyways changed the API of this function, so this clone can happened outside of this function a leave the callers the opportunity to not clone if they don't need to.